### PR TITLE
fix `ipfs help` bug #5557

### PR DESCRIPTION
**Pull request by kjzz**
fix issue #5557 

License: MIT
Signed-off-by: Kejie Zhang <601172892@qq.com>

### CODE REVIEW

File: cmd/ipfs/main.go [hunk 1]
```diff
@@ -82,13 +82,22 @@ func mainRet() int {
...
+		// Handle `ipfs --version'
+		if os.Args[1] == "--version" {
 			os.Args[1] = "version"
 		}
+
+		//Handle `ipfs help` and `ipfs help <sub-command>`
+		if os.Args[1] == "help" {
+			if len(os.Args) > 2 {
+				os.Args = append(os.Args[:1], os.Args[2:]...)
+				os.Args = append(os.Args, "-h")
```
Comment (djdv) [inline cmd/ipfs/main.go]: This should be changed to the long flag `--help` as that applies to `ipfs` itself as well as all subcommands
`-h` only works with `ipfs -h` for instance `ipfs add -h` has a different meaning, but `ipfs add --help` is what we would want.

Comment (djdv) [reply cmd/ipfs/main.go]: Disregard this, I was confusing `-h` and `-H`.
Although I am wondering which we should use for help, the short description or the long.

**Comment by djdv** [comment]
My review is squashed.
Just a note for the next person.
I'm wondering if we should use the short help description or the long one. `-h` vs `--help`.
No strong opinion either way.

**Comment by kjzz** [comment]
Thx for your quick review @djdv . 

> I'm wondering if we should use the short help description or the long one. `-h` vs `--help`.

IMO,maybe `-h` will be better.
Because ``

File: cmd/ipfs/main.go (hunk 1 above)
Comment (Kubuxu) [inline cmd/ipfs/main.go]: I would switch it to `--help` as `-h` is a short help and `--help` is a long help. (help help help :smile)

Comment (Kubuxu) [reply cmd/ipfs/main.go]: Same in line 98.

**Comment by kjzz** [comment]
@djdv @Kubuxu i have update the pr . And I have changed `-h` to `--help` ,please help me review again.Thx.

**Comment by kjzz** [comment]
@djdv @magik6k If you have any time , can you please help me review my pr ?Thx a lot.

**Comment by djdv** [comment]
Sorry for the delay @kjzz 
I wasn't very clear.
My initial review was only blocking on the question of using `-h` or `--help`.
With the decision to use `--help` I approve the changes.
But was waiting for input from someone with merge capabilities. (like @Kubuxu, who has posted above)

**Comment by kjzz** [comment]
Thx for advice @Kubuxu.I have update my pr.Please help me review it again.

**Comment by Stebalien** [comment]
This is a bit funky but, well, it was already kind of a hack. This should at least be less surprising for users.

**Pull request merged**